### ISSUE TITLE
[BUG] TYPO3 9.5: language handling in indexer is broken

### ISSUE THREAD

**Issue by dmitryd** [TYPO3-Solr/ext-solr#2295]
**Describe the bug**

In TYPO3 before 9.x it was ok to cache TSFE in the indexer using the root page id and language id like this (Classes/Util.php):

```
        $cacheId = $pageId . '|' . $language;


        if (!isset($tsfeCache[$cacheId]) || !$useCache) {
```

In TYPO3 9.x this no longer works correctly because language is not a property of TSFE anymore, it is a property of the context object. I found that records often are not indexed correctly if using `CONTENT` typoscript for rendering irre records (think of irre records in news). Even if solr thinks it tries to index them for the default language and use TSFE with language=0, context may have language id=1. The code inside `ContentContentObject` will try to apply overlay using functions from `PageRepository`. `PageRepository` uses context object and it may have value, which does not correspond to the value in TSFE. So overlay will return the empty value and record will be discarded. For me it happened like this:

* record 1 is indexed with the default language
* TSFE is created with language id=0 and cached
* context is initialized with language=0
* record 1 is indexed with language=1
* TSFE is created with language id=1 and cached
* context is initialized with language=1
* record 2 is indexed with the default language
* TSFE is *not* created but taken from cache, language id=0
* context is *not* initialized but still uses language=1 from the previous run
* ContentContentObject applies overlay with language=1 => unsets the record in the default language, no more content can be indexed

So each connected irre record gets just one single element.

**To Reproduce**
Steps to reproduce the behavior are more or less described above

 - TYPO3 Version: 9.5.5
 - EXT:solr Version: 9.0.2

** Possible fix **

I found that not disabling TSFE cache in Util.php solves the problem.

**Comment by gerchomat** [comment]
Same problem here

**Comment by gerchomat** [comment]
If you work with the old language-condition syntax and set the config.sys_language_uid it works.
`[globalVar = GP:L = 1]
    config.sys_language_uid = 1
[global]`

**Comment by SomeBdyElse** [comment]
I can confirm this bug. Adding `[globalVar = GP:L = 1] config.sys_language_uid = 1 [global]` did not help for me. Once I applied that TS condition, translated records in the translated core looked fine. However, they also started to show up in the default language Solr core. 
I can confirm the behavior that @dmitryd describes. The TSFE settings from one language seems to carry over to the next one.

**Comment by dmitryd** [comment]
Conditions will not help here because they do not affect how solr indexes records. To index records, solr fetches the default language record and then loops over languages fetching overlays. Nothing to do with conditions at all.

**Comment by dkd-kaehm** [comment]
Fixed in #2383
